parser.go

```diff
@@ -177,6 +177,50 @@ func (parser *arguments) needRoot() bool {
 	}
 }
 
+//needWait checks if waitLock() should be called before calling pacman
+func (parser *arguments) needWait() bool {
+	if parser.existsArg("h", "help") {
+		return false
+	}
+
+	if parser.existsArg("p", "print") {
+		return false
+	}
+
+	switch parser.op {
+	case "D", "database":
+		return true
+	case "F", "files":
+		if parser.existsArg("y", "refresh") {
+			return true
+		}
+		return false
+	case "R", "remove":
+		return true
+	case "S", "sync":
+		if parser.existsArg("y", "refresh") {
```

Comment:
These are redundant, they'll be caught by the default return true at the end.

Reply:
The default is return false though, I did notice needRoot does have a lot of redundancy

Reply:
The default inside of sync is true though.

Reply:
True. :innocent:

Reply:
Maybe needRoot needs a touch up as well

Reply:
Probably, been a long time since I've even looked at it.